tripy/tripy/dtype_info.py

```diff
@@ -16,35 +16,61 @@
 #
 
 import inspect
+from typing import Union, Optional, get_origin, get_args, ForwardRef
+from collections import namedtuple
 
 TYPE_VERIFICATION = {}
+FUNC_W_DOC_VERIF = []
 RETURN_VALUE = "RETURN_VALUE"
 
+class InputValues:
+    dtype: str = None
+    init: Union[tuple,list] = None
+    shape: tuple = None
+    target: str = None
+    count: int = None
 
-def dtype_info(dtype_variables: dict, dtype_constraints: dict):
-
+def dtype_info(
+    dtype_variables: dict = {},
+    dtype_constraints: dict = {},
+    param_type_specification: dict = {},
+    function_name : Optional[str] = "",
+):
+    """
+    This function is a decorator that populates TYPE_VERIFICATION global dictionary which will be used by
+    test_dtype_constraints.py to verify the dtypes of all operations.
+    """
     def decorator(func_obj):
```

Comment:
We should actually remove this complexity from the decorator. This will be run every time someone imports Tripy and for every single op. We only actually use it in the tests, so that's where the work should happen ideally. Can we make the decorator just store the function/constraints?

Reply:
I simplified it a lot. Basically `constraints.py` is only storing the inputs now.